Frontend_Web_Development_React_Angular_Vue/Vue/Week 4/List rendering_(v-for).md

```diff
@@ -0,0 +1,263 @@
+
+**HTML**
+
+    <ul id="example-2">
+      <li v-for="(item, index) in items">
+        {{ parentMessage }} - {{ index }} - {{ item.message }}
+      </li>
+    </ul>
+
+**JS**
+
+    import Vue from "vue";
+    var example2 = new Vue({
+      el: '#example-2',
+      data: {
+        parentMessage: 'Parent',
+        items: [
+          { message: 'Foo' },
+          { message: 'Bar' }
+        ]
+      }
+    })
+    
+**Result:** 
+
+![image](https://user-images.githubusercontent.com/57637086/140803990-6fbb0428-86d2-47b2-96f2-d23879ab90ee.png)
+
+
+You can also use of as the delimiter instead of in, so that it is closer to JavaScript’s syntax for iterators:
+
+    <div v-for="item of items"></div>
+    
+## v-for with an Object
+
+You can also use v-for to iterate through the properties of an object.
+
+**HTML**
+
+    <ul id="v-for-object" class="demo">
+      <li v-for="value in object">
+        {{ value }}
+      </li>
+    </ul>
+
+**JS**
+
+    import Vue from "vue";
+    new Vue({
+      el: '#v-for-object',
+      data: {
+        object: {
+          title: 'How to do lists in Vue',
+          author: 'Jane Doe',
+          publishedAt: '2016-04-10'
+        }
+      }
+    })
+    
+**Result:**
+
+![image](https://user-images.githubusercontent.com/57637086/140804226-e7fb8fbd-033b-4219-ab6d-2dddf94c371c.png)
+
+
+## You can also provide a second argument for the property’s name (a.k.a. key):
+
+
+    <div v-for="(value, name) in object">
+      {{ name }}: {{ value }}
+    </div>
+    
+   ![image](https://user-images.githubusercontent.com/57637086/140804448-1910b9b0-c8ce-49c5-aa60-4ff9db1da900.png)
+
+**And another for the index:**
+
+    <div v-for="(value, name, index) in object">
+      {{ index }}. {{ name }}: {{ value }}
+    </div>
+
+![image](https://user-images.githubusercontent.com/57637086/140804353-e20310ac-e394-4bd6-89e8-7c0e72b1c0e4.png)
+
+
+
+
+    
+## Vue.js - Display a list of items with v-for    
+
+**HTML**
+
+    <!DOCTYPE html>
+    <html lang="en">
+    <head>
+        <meta charset="UTF-8">
+        <title>Vue.js - Display a list of items with v-for</title>
+        <link href="//netdna.bootstrapcdn.com/bootstrap/4.5.2/css/bootstrap.min.css" rel="stylesheet" />
+    </head>
+    <body>
+        <div id="app"></div>
+
+    </body>
+    </html>
+
+**JS**
+
+    import Vue from "vue";
+
+    import App from "./App";
+
+    new Vue({
+        el: "#app",
+        render: h => h(App)
+    });
+
+**App.vue**
+
+    <template>
+        <div class="container">
+            <h3 class="p-3 text-center">Vue.js - Display a list of items with v-for</h3>
+            <table class="table table-striped table-bordered">
+                <thead>
+                    <tr>
+                        <th>Name</th>
+                        <th>Email</th>
+                        <th>Role</th>
+                    </tr>
+                </thead>
+                <tbody>
+                    <tr v-for="user in users" :key="user.id">
```

Comment:
There is nothing with a key of "id" in your "user" object.
Correct this with a valid key or use an index (as because index also works as a key when we don't have any id's).